xtask/src/stationpedia.rs

```diff
@@ -0,0 +1,346 @@
+use serde_derive::{Deserialize, Serialize};
+use serde_with::{serde_as, DisplayFromStr};
+use std::collections::BTreeMap;
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(rename = "Stationpedia", deny_unknown_fields)]
+pub struct Stationpedia {
+    pub pages: Vec<Page>,
+    pub reagents: BTreeMap<String, Reagent>,
+    #[serde(rename = "scriptCommands")]
+    pub script_commands: BTreeMap<String, Command>,
+}
+
+#[allow(dead_code)]
+impl Stationpedia {
+    pub fn lookup_prefab_name(&self, prefab_name: &'_ str) -> Option<&Page> {
+        self.pages.iter().find(|p| p.prefab_name == prefab_name)
+    }
+
+    pub fn lookup_key(&self, key: &str) -> Option<&Page> {
+        self.pages.iter().find(|p| p.key == key)
+    }
+
+    pub fn lookup_hash(&self, hash: i32) -> Option<&Page> {
+        self.pages.iter().find(|p| p.prefab_hash == hash)
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, PartialOrd, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct Reagent {
+    #[serde(rename = "Hash")]
+    pub hash: i64,
+    #[serde(rename = "Unit")]
+    pub unit: String,
+    #[serde(rename = "Sources")]
+    pub sources: Option<BTreeMap<String, f64>>,
+}
+
+#[derive(Clone, Debug, PartialEq, PartialOrd, Eq, Ord, Hash, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct Command {
+    pub desc: String,
+    pub example: String,
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct Page {
+    #[serde(rename = "ConnectionInsert")]
+    pub connection_insert: Vec<ConnectionInsert>,
+    #[serde(rename = "ConstructedByKits")]
+    pub constructs: Vec<Constructs>,
+    #[serde(rename = "Description")]
+    pub description: String,
+    #[serde(rename = "Device")]
+    pub device: Option<Device>,
+    /// the item , if none then deprecated
```

Comment:
My comment is wrong i think, its none if not an item. I even use that fact in https://github.com/Emilgardis/Stationeers-Wiki-Page-Helper/blob/64b21f49390ceede70d519d3dd20931a67a3a4b6/src/commands/wikibox.rs#L134